### PR TITLE
Add taplo.toml and format tomls

### LINKED ISSUE #2766: Format toml in CI
I'd like us to add a toml formatter to CI. It'd be nice to not have to deal with formatting or alphabetical sorting in our `Cargo.toml`'s.

### PR DESCRIPTION
**Pull request by lann**
This is a pretty conservative config mostly impacting dependency ordering.

All formatting in this PR was from running `taplo fmt`.

Fixes #2766

### CODE REVIEW

**Comment by lann** [comment]
On hold until `factors` merges to avoid conflicts.

File: taplo.toml
Comment (radu-matei) [inline taplo.toml:15]: I am strongly in favor of this.

**Review by radu-matei — APPROVED** [review]
Is the intention to also check formatting in CI?

**Comment by rylev** [comment]
Given how low of the priority list this, I'd be hesitant to check this in CI. This would just be yet another barrier for new contributors to have to get over (although I realize editing Cargo.toml files is somewhat rare).

**Review by rylev — APPROVED** [review]

**Pull request merged**
